tough/src/editor.rs

```diff
@@ -0,0 +1,744 @@
+        };
+
+        let path = outdir.join(filename);
+        std::fs::write(&path, &self.buffer).context(error::FileWrite { path })
+    }
+}
+
+/// A set of signed TUF Repository metadata.
+///
+/// This metadata represents a signed TUF repository and provides the ability
+/// to write the metadata to disk.
+///
+/// Note: without the target files, the repository cannot be used. It is up
+/// to the user to ensure all the target files referenced by the metadata are
+/// available. There are convenience methods to help with this.
+#[derive(Debug)]
+pub struct SignedRepository {
+    root: SignedRole<Root>,
+    targets: SignedRole<Targets>,
+    snapshot: SignedRole<Snapshot>,
+    timestamp: SignedRole<Timestamp>,
+}
+
+impl SignedRepository {
+    /// Writes the metadata to the given directory. If consistent snapshots
+    /// are used, the appropriate files are prefixed with their version.
+    pub fn write<P>(&self, outdir: P) -> Result<()>
+    where
+        P: AsRef<Path>,
+    {
+        let consistent_snapshot = self.root.signed.signed.consistent_snapshot;
+        self.root.write(&outdir, consistent_snapshot)?;
+        self.targets.write(&outdir, consistent_snapshot)?;
+        self.snapshot.write(&outdir, consistent_snapshot)?;
+        self.timestamp.write(&outdir, consistent_snapshot)
+    }
+
+    /// Crawls a given directory and symlinks any targets found to the given
+    /// "out" directory. If consistent snapshots are used, the target files
+    /// are prefixed with their `sha256`.
+    ///
+    /// For each file found in the `indir`, the method gets the filename and
+    /// if the filename exists in `Targets`, the file's sha256 is compared
+    /// against the data in `Targets`. If this data does not match, the
+    /// method will fail. If all is well, the target is symlinked.
+    pub fn link_targets<P1, P2>(&self, indir: P1, outdir: P2) -> Result<()>
+    where
+        P1: AsRef<Path>,
+        P2: AsRef<Path>,
+    {
+        let outdir = outdir.as_ref();
+        std::fs::create_dir_all(outdir).context(error::DirCreate { path: outdir })?;
+        // Get the absolute path of the indir and outdir
+        let abs_indir = std::fs::canonicalize(&indir).context(error::AbsolutePath {
+            path: indir.as_ref().to_owned(),
+        })?;
+        let abs_outdir =
+            std::fs::canonicalize(outdir).context(error::AbsolutePath { path: outdir })?;
+        let repo_targets = &self.targets.signed.signed.targets;
+
+        // Walk the absolute path of the indir. Using the absolute path here
+        // means that `entry.path()` call will return its absolute path.
+        let walker = WalkDir::new(&abs_indir).follow_links(true);
+        for entry in walker {
+            let entry = entry.context(error::WalkDir {
+                directory: &abs_indir,
+            })?;
+
+            // If the entry is a file...
+            if !entry.file_type().is_file() {
+                continue;
+            } else {
```

Comment:
```suggestion
                continue;
            }
```
You don't need all the code below to be indented for this.  The `continue` will move to the next iteration and skip it all.

Reply:
😳 

Sorry about that... This is a silly mistake.